docs/tasks.md

```diff
@@ -0,0 +1,146 @@
+Background Tasks in AWX
+=======================
+
+AWX runs a lot of Python code asynchronously _in the background_ - meaning
+_outside_ of the context of an HTTP request, such as:
+
+* Any time a Job is launched in AWX (a Job Template, an Adhoc Command, a Project
+  Update, an Inventory Update, a System Job), a background process retrieves
+  metadata _about_ that job from the database and forks some process (e.g.,
+  `ansible-playbook`, `awx-manage inventory_import`) 
+* Certain expensive or time consuming tasks run in the background
+  asynchronously (like deleting an inventory).
+* AWX runs a variety of periodic background tasks on a schedule.  Some examples
+  are:
+    - AWX's "Task Manager/Scheduler" wakes up periodically and looks for
+      `pending` jobs that have been launched and are ready to start running.
+    - AWX periodically runs code that looks for scheduled jobs and launches
+      them.
+    - AWX runs a variety of periodic tasks that clean up temporary files, and
+      perform various administrative checks
+    - Every node in an AWX cluster runs a periodic task that serves as
+      a heartbeat and capacity check
+
+Tasks, Queues and Workers
+----------------
+
+To accomplish this, AWX makes use of a "Task Queue" abstraction.  Task Queues
+are used as a mechanism to distribute work across machines in an AWX
+installation.  A Task Queue's input is a unit of work called a Task. Dedicated
+worker processes running on every AWX node constantly monitor these queues for
+new work to perform.
+
+AWX communicates with these worker processes via AMQP - using RabbitMQ,
+specifically - to mediate between clients and workers. To initiate a task, the
+client (generally, Python code in the AWX API) publishes a message to a queue,
+and RabbitMQ then delivers that message to one or more workers.
+
+Clustered AWX installations consist of multiple workers spread across every
+node, giving way to high availability and horizontal scaling.
+
+Direct vs Fanout Messages
+-------------------------
+
+AWX publishes tasks in two distinct ways.
+
+*Direct* messages are bound _directly_ to a specific named queue.  When you launch
+a Job Template in AWX, it looks at the available capacity of the various nodes
+in your cluster and chooses an `Execution Node` where the playbook will run.
+In this scenario, AWX publishes a message to a direct queue associated with
+that AWX node.  The worker processes running on that AWX node are specifically
+bound to _listen_ for new events on their named queue.
```

Comment:
The worker processes listen for new events on the queue named for the _instance_?

Reply:
Yea, that's probably a clearer way to describe that.